### DISCUSSION TITLE
Review of Languagetool

### DISCUSSION THREAD

**Discussion by AlexeyMatveev686** [ONLYOFFICE/onlyoffice.github.io#245] [Polls]
<img src="https://raw.githubusercontent.com/ONLYOFFICE/onlyoffice.github.io/master/discussions-icons/languagetool.svg">

**Comment by GTP95** [comment]
It does its job, but doesn't integrate well with Presentation: the user is required to manually select the text to be checked or to type it inside a textbox. The plugin should instead be able to automatically read the text from the slides.

**Comment by nathschlaeger** [comment]
The integration is poor and for long text format like typoscripts eg novels not usable. It would be good if Languagtool could be integrated in OnlyOffice like it is being integrated in LibreOffice.

**Comment by Mannshoch** [comment]
I have a self-hosted Languagetool. I do not find a way to change the Server URL.

**Reply by tjaeuth** [reply]
In the README they mention to use the `serviceUrl` variable: https://github.com/ONLYOFFICE/onlyoffice.github.io/blob/master/sdkjs-plugins/content/languagetool/README.md

**Comment by JeffersonLeung** [comment]
The lack of in-line editing is horrible. Copy and pasting text isn't viable. Normally my text is formatted and structured in a certain way. Cut and pasting screws it up. This is not a praticle method for grammar check at all.

**Comment by Misokados** [comment]
Hello,
We're checking the mentioned requests. We will update this thread as soon as possible.

**Comment by Misokados** [comment]
Dear users,
We have found an issue with the current version of the plugin. It will be fixed as soon as possible in the upcoming version 9.3. Once version 9.3 is released, please recheck the situation.

**Comment by sayurisimmers** [comment]
I find the Language Tool in OnlyOffice very buggy; sometimes I have to select "check" several times before anything happens, and most of the time it doesn't work and just keeps checking. I don't know if it's related to the text size, but it could have indicated that to make it easier. And sometimes I have to create text multiple times by clicking "check," and it's as if I hadn't pressed anything.

**Reply by Misokados** [reply]
Hello @sayurisimmers 
We just have released the Desktop app v.9.3. Please update the app and re-check the situation. 
If the issue still occurs, please record a video file.
